portfolio/src/main/webapp/script.js

```diff
@@ -1,24 +1,21 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 function getMessages() {
   fetch('/data').then(response => response.json()).then((messages) => {
     const messagesElement = document.getElementById('messages-container');
     messagesElement.innerHTML = "";
     console.log(messages);
     for (i of messages) {
-      messagesElement.innerHTML += "<p>"+i.message+"</p><br>";
+      if (i.score == -2) {
+        messagesElement.appendChild(createListElement(i.message + " Score: n/a"));
+      }
+      else {
+        messagesElement.appendChild(createListElement(i.message + " Score: " + i.score));
+      }
```

Comment:
One principle to try and follow is to reduce (as much as possible) the amount of duplicated code. Note how lines 8 and 11 are almost identical: `messagesElement.appendChild(createListElement(i.message + <something>))`. So one thing to consider is could precompute the `<something>` and then only do the `appendChild` call once? I'd do something like this:

```
const printableScore = (i.score == -2 ? "n/a" : i.score);
messages.appendChild(createListElement(i.message + " Score: " + printableScore));
```

(Note: Javascript is not my primary language, but I'm certain that something like this is possible).

Why is this principle a good idea? It generally leads to code that is more easily maintained. Imagine if you wanted to change the text to be "The score for this message is: "...then this change must be made accurately in two places, which is error prone (could be forgotten, could introduce typos, etc.). Another way of looking at it is "a single thing is always consistent with itself)".

Reply:
Very good point!